src/branch_view.js

```diff
@@ -202,7 +214,12 @@ branch_view.initJourney = function(branch_key, data, eventData, options, branch)
 	if (branchViewData && !hideJourney && !no_journeys) {
 		journeys_utils.branchViewId = branchViewData.id;
 		branch['renderQueue'](function() {
-			requestData = compileRequestData(branch);
+			var makeNewLink = options && options['make_new_link'] || false;
+			var linkClickId = null;
+			if (!makeNewLink) {
+				linkClickId = fetchLinkClickId(branch._referringLink());
+			}
+			requestData = compileRequestData(branch, linkClickId);
```

Comment:
I would just pass in `makeNewLink` instead of `linkClickId` and handle the logic above in the `compileRequestData` function